### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by AlexDochioiu**
* Improved BLE reliability
* Serializing the requests to avoid running them in parallel

## Release Notes

- **New Features**
  - Introduced new utility for hexadecimal encoding and decoding.
  - Added new classes for managing ledger operations, including `LedgerSimpleOperation` and `LedgerComplexOperation`.
  - Enhanced connection management with a new `RequestQueue` for better operation handling.

- **Bug Fixes**
  - Improved error handling in connection management, replacing `ConnectionTimeoutException` with `EstablishConnectionException`.

- **Refactor**
  - Updated method signatures across various components to support new operation types and improve clarity.
  - Marked `LedgerOperation` as deprecated, recommending the use of `LedgerSimpleOperation` or `LedgerComplexOperation`.

- **Documentation**
  - Updated documentation to reflect changes in operation handling and new utility features.

### CODE REVIEW

File: lib/src/operations/ledger_simple_operation.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    if (prependDataLength) {
+      writer.writeUint8(data.length);
+    }
+    if (data.isNotEmpty) {
+      writer.write(data);
+    }
+
+    if (debugPrintEnabled) {
+      // ignore: avoid_print
+      print("$debugName: ${hex.encode(writer.toBytes())}");
```
Comment (konstantinullrich) [inline lib/src/operations/ledger_simple_operation.dart:42]: Why not replace `$debugName` with `$runtimeType`?

**Review by konstantinullrich — APPROVED** [review]

File: lib/src/operations/ledger_simple_operation.dart (hunk 1 above)
Comment (AlexDochioiu) [reply lib/src/operations/ledger_simple_operation.dart:42]: In the Cardano plugin, I actually don't create a new class extending `LedgerSimpleOperation`. I just directly create an instance of it. In such a case, `LedgerSimpleOperation` would be the runtimeType for all of them.

Comment (AlexDochioiu) [reply lib/src/operations/ledger_simple_operation.dart:42]: I can just make `debugName` optional and do use `debugName ?? runtimeType` 🤔

Comment (konstantinullrich) [reply lib/src/operations/ledger_simple_operation.dart:42]: That would be ideal

Comment (AlexDochioiu) [reply lib/src/operations/ledger_simple_operation.dart:42]: Done. I updated to make `debugName` optional in constructor.

**Review by konstantinullrich — APPROVED** [review]

**Pull request merged**